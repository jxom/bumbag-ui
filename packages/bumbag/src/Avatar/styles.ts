import { css, cssClass } from '../styled';
import { palette, theme } from '../utils';

export const Avatar = (styleProps) => cssClass`
  width: 60px;
  height: 60px;
  overflow: hidden;
  object-fit: cover;

  ${
    styleProps.variant === 'circle' &&
    css`
      border-radius: 50%;

      & {
        ${theme(styleProps.themeKey, `circle.base`)(styleProps)};
      }
    `
  }

  ${
    styleProps.initials &&
    css`
      background-color: ${palette(styleProps.palette)(styleProps)};
      color: ${palette(`${styleProps.palette}Inverted`)(styleProps)};
      display: flex;
      justify-content: center;
      align-items: center;
    `
  }

  ${getSizeAttributes(styleProps)};

  & {
    ${theme(styleProps.themeKey, `styles.base`)(styleProps)};
  }
`;

function getSizeAttributes(styleProps) {
  const sizeAttributes = {
    small: css`
      width: 40px;
      height: 40px;

      & {
        ${theme(styleProps.themeKey, `styles.sizes.small`)(styleProps)};
      }
    `,
    default: css`
      width: 60px;
      height: 60px;
      font-size: 24px;

      & {
        ${theme(styleProps.themeKey, `styles.sizes.default`)(styleProps)};
      }
    `,
    medium: css`
      width: 80px;
      height: 80px;
      font-size: 36px;

      & {
        ${theme(styleProps.themeKey, `styles.sizes.medium`)(styleProps)};
      }
    `,
    large: css`
      width: 100px;
      height: 100px;
      font-size: 48px;

      & {
        ${theme(styleProps.themeKey, `styles.sizes.large`)(styleProps)};
      }
    `,
  };
  return (
    sizeAttributes[styleProps.size] ||
    css`
      width: ${styleProps.size};
      height: ${styleProps.size};
    `
  );
}
