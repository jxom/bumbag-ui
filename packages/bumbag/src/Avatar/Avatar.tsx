import { Box as ReakitBox } from 'reakit';

import { Palette, Omit, Size, Flexible } from '../types';
import { useClassName, createComponent, createElement, createHook } from '../utils';
import { Box } from '../Box';
import { Image, ImageProps } from '../Image';

import * as styles from './styles';

export type LocalAvatarProps = {
  /** Variant of the avatar. */
  variant?: Flexible<'circle', string>;
  /** URL of the avatar image. */
  src?: string;
  /** Size of the avatar. */
  size?: Size | string;
  /** Initials to place on the avatar. */
  initials?: string;
  /** Color of the avatar. */
  palette?: Palette;
};
export type AvatarProps = Omit<ImageProps, 'src'> & LocalAvatarProps;

const useProps = createHook<AvatarProps>(
  (props = {}, { themeKey, themeKeyOverride }) => {
    const imageProps = Image.useProps({ ...props }, { themeKey: 'Avatar' });

    const className = useClassName({
      style: styles.Avatar,
      styleProps: props,
      themeKey,
      themeKeyOverride,
      prevClassName: imageProps.className,
    });

    let children;
    if (props.initials) {
      children =
        props.initials.split(' ').length === 2 ? props.initials.match(/\b\w/g) : props.initials.substring(0, 2);
    }

    return { ...imageProps, className, children };
  },
  {
    defaultProps: {
      size: 'default',
    },
    themeKey: 'Avatar',
  }
);

export const Avatar = createComponent<AvatarProps>(
  (props) => {
    const imageProps = useProps(props);

    let use = props.use;
    if (props.initials) {
      use = Box;
    }

    return createElement({ component: ReakitBox, use, htmlProps: imageProps });
  },
  {
    attach: {
      useProps,
      displayName: 'Avatar',
    },
    defaultProps: {
      use: 'img',
    },
    themeKey: 'Avatar',
  }
);
