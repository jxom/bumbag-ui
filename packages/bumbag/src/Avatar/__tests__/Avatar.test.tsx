import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Avatar } from '../Avatar';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Avatar src="test.png" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(<Avatar alt="Bean" backgroundColor="red" src="test.png" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with variant circle', () => {
    const { container } = render(<Avatar alt="Bean" variant="circle" src="test.png" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  ['small', 'default', 'medium', 'large'].forEach((size) => {
    it(`should render correctly for size ${size}`, () => {
      const { container } = render(<Avatar alt="Bean" size={size} src="test.png" />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('should render correctly for initials', () => {
    const { container } = render(<Avatar initials="JM" palette="primary" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('overrides', () => {
  it('Avatar.base should render correctly', () => {
    const { container } = render(
      <Avatar src="test.png" overrides={{ Avatar: { styles: { base: { backgroundColor: 'red' } } } }} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Avatar.circle.base should render correctly', () => {
    const { container } = render(
      <Avatar src="test.png" overrides={{ Avatar: { styles: { circle: { backgroundColor: 'red' } } } }} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Avatar.sizes.small should render correctly', () => {
    const { container } = render(
      <Avatar
        src="test.png"
        size="small"
        overrides={{ Avatar: { styles: { sizes: { small: { backgroundColor: 'red' } } } } }}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Avatar.sizes.default should render correctly', () => {
    const { container } = render(
      <Avatar src="test.png" overrides={{ Avatar: { styles: { sizes: { default: { backgroundColor: 'red' } } } } }} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Avatar.sizes.medium should render correctly', () => {
    const { container } = render(
      <Avatar
        src="test.png"
        size="medium"
        overrides={{ Avatar: { styles: { sizes: { medium: { backgroundColor: 'red' } } } } }}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Avatar.sizes.large should render correctly', () => {
    const { container } = render(
      <Avatar
        src="test.png"
        size="large"
        overrides={{ Avatar: { styles: { sizes: { large: { backgroundColor: 'red' } } } } }}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('Avatar.base should render correctly', () => {
    const { container } = render(<Avatar src="test.png" />, {
      theme: { Avatar: { styles: { base: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Avatar.circle.base should render correctly', () => {
    const { container } = render(<Avatar src="test.png" />, {
      theme: { Avatar: { styles: { circle: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Avatar.sizes.small should render correctly', () => {
    const { container } = render(<Avatar src="test.png" size="small" />, {
      theme: { Avatar: { styles: { sizes: { small: { backgroundColor: 'red' } } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Avatar.sizes.default should render correctly', () => {
    const { container } = render(<Avatar src="test.png" size="default" />, {
      theme: { Avatar: { styles: { sizes: { default: { backgroundColor: 'red' } } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Avatar.sizes.medium should render correctly', () => {
    const { container } = render(<Avatar src="test.png" size="medium" />, {
      theme: { Avatar: { styles: { sizes: { medium: { backgroundColor: 'red' } } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Avatar.sizes.large should render correctly', () => {
    const { container } = render(<Avatar src="test.png" size="large" />, {
      theme: { Avatar: { styles: { sizes: { large: { backgroundColor: 'red' } } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(<Avatar src="test.png" />, {
      theme: { Avatar: { defaultProps: { className: 'test', color: 'primary' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
