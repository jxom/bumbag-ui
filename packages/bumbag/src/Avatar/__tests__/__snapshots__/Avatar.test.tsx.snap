// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultProps should render correctly for className 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 60px;
  height: 60px;
  overflow: hidden;
  object-fit: cover;
  width: 60px;
  height: 60px;
  font-size: 24px;
}

.emotion-3 {
  color: #574feb !important;
}

<img
  class="emotion-0 emotion-1 emotion-2 emotion-3 test fp-Box fp-Avatar fp-Image"
  src="test.png"
/>
`;

exports[`overrides Avatar.base should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-1.emotion-1 {
  background-color: #ff0000;
}

.emotion-0 {
  width: 60px;
  height: 60px;
  overflow: hidden;
  object-fit: cover;
  width: 60px;
  height: 60px;
  font-size: 24px;
}

.emotion-0.emotion-0 {
  background-color: #ff0000;
}

<img
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Avatar fp-Image"
  src="test.png"
/>
`;

exports[`overrides Avatar.circle.base should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 60px;
  height: 60px;
  overflow: hidden;
  object-fit: cover;
  width: 60px;
  height: 60px;
  font-size: 24px;
}

<img
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Avatar fp-Image"
  src="test.png"
/>
`;

exports[`overrides Avatar.sizes.default should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 60px;
  height: 60px;
  overflow: hidden;
  object-fit: cover;
  width: 60px;
  height: 60px;
  font-size: 24px;
}

.emotion-0.emotion-0 {
  background-color: #ff0000;
}

<img
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Avatar fp-Image"
  src="test.png"
/>
`;

exports[`overrides Avatar.sizes.large should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 60px;
  height: 60px;
  overflow: hidden;
  object-fit: cover;
  width: 100px;
  height: 100px;
  font-size: 48px;
}

.emotion-0.emotion-0 {
  background-color: #ff0000;
}

<img
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Avatar fp-Image"
  src="test.png"
/>
`;

exports[`overrides Avatar.sizes.medium should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 60px;
  height: 60px;
  overflow: hidden;
  object-fit: cover;
  width: 80px;
  height: 80px;
  font-size: 36px;
}

.emotion-0.emotion-0 {
  background-color: #ff0000;
}

<img
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Avatar fp-Image"
  src="test.png"
/>
`;

exports[`overrides Avatar.sizes.small should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 60px;
  height: 60px;
  overflow: hidden;
  object-fit: cover;
  width: 40px;
  height: 40px;
}

.emotion-0.emotion-0 {
  background-color: #ff0000;
}

<img
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Avatar fp-Image"
  src="test.png"
/>
`;

exports[`props should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 60px;
  height: 60px;
  overflow: hidden;
  object-fit: cover;
  width: 60px;
  height: 60px;
  font-size: 24px;
}

<img
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Avatar fp-Image"
  src="test.png"
/>
`;

exports[`props should render correctly for initials 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2 {
  width: 60px;
  height: 60px;
  overflow: hidden;
  object-fit: cover;
  background-color: #574feb;
  color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 60px;
  height: 60px;
  font-size: 24px;
}

<div
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Avatar fp-Image"
>
  JM
</div>
`;

exports[`props should render correctly for size default 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 60px;
  height: 60px;
  overflow: hidden;
  object-fit: cover;
  width: 60px;
  height: 60px;
  font-size: 24px;
}

<img
  alt="Bean"
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Avatar fp-Image"
  src="test.png"
/>
`;

exports[`props should render correctly for size large 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 60px;
  height: 60px;
  overflow: hidden;
  object-fit: cover;
  width: 100px;
  height: 100px;
  font-size: 48px;
}

<img
  alt="Bean"
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Avatar fp-Image"
  src="test.png"
/>
`;

exports[`props should render correctly for size medium 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 60px;
  height: 60px;
  overflow: hidden;
  object-fit: cover;
  width: 80px;
  height: 80px;
  font-size: 36px;
}

<img
  alt="Bean"
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Avatar fp-Image"
  src="test.png"
/>
`;

exports[`props should render correctly for size small 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 60px;
  height: 60px;
  overflow: hidden;
  object-fit: cover;
  width: 40px;
  height: 40px;
}

<img
  alt="Bean"
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Avatar fp-Image"
  src="test.png"
/>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 60px;
  height: 60px;
  overflow: hidden;
  object-fit: cover;
  width: 60px;
  height: 60px;
  font-size: 24px;
}

.emotion-3 {
  background-color: #ff0000 !important;
}

<img
  alt="Bean"
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Avatar fp-Image"
  src="test.png"
/>
`;

exports[`props should render correctly with variant circle 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 60px;
  height: 60px;
  overflow: hidden;
  object-fit: cover;
  border-radius: 50%;
  width: 60px;
  height: 60px;
  font-size: 24px;
}

<img
  alt="Bean"
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Avatar fp-Image"
  src="test.png"
/>
`;

exports[`theming Avatar.base should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-1.emotion-1 {
  background-color: #ff0000;
}

.emotion-0 {
  width: 60px;
  height: 60px;
  overflow: hidden;
  object-fit: cover;
  width: 60px;
  height: 60px;
  font-size: 24px;
}

.emotion-0.emotion-0 {
  background-color: #ff0000;
}

<img
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Avatar fp-Image"
  src="test.png"
/>
`;

exports[`theming Avatar.circle.base should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 60px;
  height: 60px;
  overflow: hidden;
  object-fit: cover;
  width: 60px;
  height: 60px;
  font-size: 24px;
}

<img
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Avatar fp-Image"
  src="test.png"
/>
`;

exports[`theming Avatar.sizes.default should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 60px;
  height: 60px;
  overflow: hidden;
  object-fit: cover;
  width: 60px;
  height: 60px;
  font-size: 24px;
}

.emotion-0.emotion-0 {
  background-color: #ff0000;
}

<img
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Avatar fp-Image"
  src="test.png"
/>
`;

exports[`theming Avatar.sizes.large should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 60px;
  height: 60px;
  overflow: hidden;
  object-fit: cover;
  width: 100px;
  height: 100px;
  font-size: 48px;
}

.emotion-0.emotion-0 {
  background-color: #ff0000;
}

<img
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Avatar fp-Image"
  src="test.png"
/>
`;

exports[`theming Avatar.sizes.medium should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 60px;
  height: 60px;
  overflow: hidden;
  object-fit: cover;
  width: 80px;
  height: 80px;
  font-size: 36px;
}

.emotion-0.emotion-0 {
  background-color: #ff0000;
}

<img
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Avatar fp-Image"
  src="test.png"
/>
`;

exports[`theming Avatar.sizes.small should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 60px;
  height: 60px;
  overflow: hidden;
  object-fit: cover;
  width: 40px;
  height: 40px;
}

.emotion-0.emotion-0 {
  background-color: #ff0000;
}

<img
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Avatar fp-Image"
  src="test.png"
/>
`;
